plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/ui/AccountSelector.java

```diff
@@ -65,6 +65,9 @@ public AccountSelector(Composite parent, IGoogleApiFactory apiFactory) {
     GridDataFactory.fillDefaults().grab(true, false).applyTo(accountEmailComposite);
     GridLayoutFactory.fillDefaults().generateLayout(accountEmailComposite);
     GridLayoutFactory.fillDefaults().generateLayout(this);
+    apiFactory.addCredentialChangeListener(() -> {
```

Comment:
Should this also have an associated `remove` call? What prevents this listener from living longer than the selector's usefulness?

Reply:
I added a dispose listener